sdk/storage/Azure.Storage.Files.Shares/tests/ShareClientTests.cs

```diff
@@ -734,6 +735,51 @@ public async Task Delete_Error()
                 e => Assert.AreEqual("ShareNotFound", e.ErrorCode));
         }
 
+        [Test]
+        [ServiceVersion(Min = ShareClientOptions.ServiceVersion.V2019_12_12)]
+        public async Task RestoreAsync()
+        {
+            // Arrange
+            ShareServiceClient service = GetServiceClient_SoftDelete();
+            string shareName = GetNewShareName();
+            ShareClient share = InstrumentClient(service.GetShareClient(shareName));
+            await share.CreateAsync();
+            await share.DeleteAsync();
+            IList<ShareItem> shares = await service.GetSharesAsync(states: ShareStates.Deleted).ToListAsync();
+            ShareItem shareItem = shares.Where(s => s.Name == shareName).FirstOrDefault();
+
+            // It takes some time for the Share to be deleted.
+            if (Mode != RecordedTestMode.Playback)
+            {
+                await Task.Delay(30000);
+            }
+
+            // Act
+            await share.RestoreAsync(
+                shareItem.Name,
+                shareItem.Version);
+
+            // Assert
+            await share.GetPropertiesAsync();
+
+            // Cleanup
+            await share.DeleteAsync();
+        }
+
+        [Test]
+        [ServiceVersion(Min = ShareClientOptions.ServiceVersion.V2019_12_12)]
+        public async Task RestoreAsync_Error()
+        {
+            // Arrange
+            ShareServiceClient service = GetServiceClient_SoftDelete();
+            ShareClient share = InstrumentClient(service.GetShareClient(GetNewShareName()));
+
+            // Act
+            await TestHelper.AssertExpectedExceptionAsync<RequestFailedException>(
+                share.RestoreAsync(GetNewShareName(), "01D60F8BB59A4652"),
```

Comment:
nit: can you introduce descriptive constant or variable for that string?

Reply:
fixed.